### PR TITLE
Consistency Across Forms

### LINKED ISSUE #208: enhancement: Ensure consistency across forms
Ensure consistency across all of the forms on the frontend, including stylistic consistency, usage of react-hook-form and Zod, folder organization, etc.

### PR DESCRIPTION
**Pull request by shane-jung**
## Changes
- Improves stylistic consistency across entrypoint editing and garden editing forms. 
- Creates a new hook `useCreateGardenAndDOI` which centralizes all of the logic for creating a Garden and it's associated DOI. Ideally, the backend will eventually handle the logic of DOI creation and publishing, but for now this is on the front end. This hook will be useful in other places in the app when we need to create a Garden in one fell swoop (like when creating a Garden in the Modal Upload page, in an upcoming PR)
- Creates a new component SelectEntrypointsTable that allows users to select/remove entrypoints from a given Garden, used both when creating a garden and when editing an existing garden. Also adds an indicator to tell a user when they can't edit the entrypoints on a Garden (aligning with new restrictions on the backend where published Gardens can't have their entrypoints edited)

- Reduces complexity in the way react-hook-form/Zod handles fields like `authors`, `contributors`, `tags`, etc. (fields that use a MultipleSelect), due to me finding a simpler way to do the validation, ensuring consistency across all forms.

### Stylistic Consistency Across Forms:
image
image

### Example of Select Entrypoints Table (Unpublished Garden vs. Published)
image
image

### CODE REVIEW

File: garden/src/api/gardens/useCreateGardenAndDOI.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { useUpdateDOI } from "../doi/useUpdateDOI";
+import { GardenCreateRequest } from "../types";
+import { useCreateGarden } from "./useCreateGarden";
+
+export const useCreateGardenAndDOI = () => {
+  const { mutateAsync: createDOI } = useCreateDOI();
+  const { mutateAsync: createGarden } = useCreateGarden();
+  const { mutateAsync: updateDOI } = useUpdateDOI();
+
+  const createGardenAndDOI = async (values: GardenCreateRequest) => {
```
Comment (WillEngler) [inline garden/src/api/gardens/useCreateGardenAndDOI.ts:11]: Until we get DOI creation into the backend garden/entrypoint creation routes, abstracting the two operations into this helper is a great idea.

File: garden/src/components/form/garden/create/schemas.ts [hunk 2]
```diff
@@ -28,20 +21,16 @@ export const formSchema = z.object({
...
+  tags: z.array(z.string()),
 
-  entrypoint_ids: z.array(
-    z.object({
-      doi: z.string(),
-      title: z.string(),
-      description: z.string().optional().nullable(),
-    }),
-  ),
+  entrypoint_ids: z.array(z.string()),
```
Comment (WillEngler) [inline garden/src/components/form/garden/create/schemas.ts:28]: I like this simplification

File: garden/src/components/form/garden/create/FormFields.tsx [hunk 3]
```diff
@@ -346,9 +273,17 @@ const Step3 = () => {
...
             <FormControl>
-              <MultipleSelector {...field} placeholder="Add authors" creatable />
+              <MultipleSelector
+                {...field}
+                placeholder="Add authors"
+                creatable
+                onChange={(value) => {
+                  field.onChange(value.map((v: any) => v.value));
+                }}
+                value={field.value.map((v: any) => ({ value: v, label: v }))}
```
Comment (WillEngler) [inline garden/src/components/form/garden/create/FormFields.tsx:285]: Seems like we need this mapping boilerplate in all the places we have array-valued fields? Looks like the right move, just a little surprising that the form library doesn't handle it more natively

**Review by WillEngler — APPROVED** [review]
Looks good to me! Thanks for being proactive on keeping things consistent across the app. Comments are just me thinking out loud as I keep coming up on this repo

File: garden/src/components/form/garden/create/FormFields.tsx (hunk 3 above)
Comment (shane-jung) [reply garden/src/components/form/garden/create/FormFields.tsx:285]: This is the best way I've found-- it's definitely possible there's a better way, I'm still finding new ways to do stuff

**Pull request merged**